Sources/NIOCore/Docs.docc/loops-futures-concurrency.md

```diff
@@ -0,0 +1,176 @@
+# EventLoops, EventLoopFutures, and Swift Concurrency
+
+This article aims to communicate how NIO's ``EventLoop``s and ``EventLoopFuture``s interact with the Swift 6
+concurrency model, particularly regarding data-race safety. It aims to be a reference for writing correct
+concurrent code in the NIO model.
+
+NIO predates the Swift concurrency model. As a result, several of NIO's concepts are not perfect matches to
+the concepts that Swift uses, or have overlapping responsibilities.
+
+## Isolation domains and executors
+
+First, a quick recap. The core of Swift 6's data-race safety protection is the concept of an "isolation
+domain". Some valuable reading regarding the concept can be found in
+[SE-0414 (Region based isolation)](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0414-region-based-isolation.md)
+but at a high level an isolation domain can be understood to be a region within which there cannot be
+multiple executors executing code at the same time.
+
+In standard Swift Concurrency, the main boundaries of isolation regions are actors and tasks. Each actor,
+including global actors, defines an isolation domain. Additionally, for functions and methods that are
+not isolated to an actor, the `Task` within which that code executes defines an isolation domain. Passing
+values between these isolation domains requires that these values are either `Sendable` (safe to hold in
+multiple domains), or that the `sending` keyword is used to force the value to be passed from one domain
+to another.
+
+A related concept to an "isolation domain" is an "executor". Again, useful reading can be found in
+[SE-0392 (Custom actor executors)](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0392-custom-actor-executors.md).
+At a high level, an executor is simply an object that is capable of executing Swift `Task`s. Executors can be
```

Comment:
Probably a nit but I'm not sure if `Task` is 100% correct here. Like there's no way to retrieve a `let theTask: Task` from a child task. But then, child tasks still ask `Task.isCancelled` so maybe it is correct :).

I would've said that they can run Jobs but then that's not really terminology that's actually used. Probably fine

Reply:
I tried for a while to rewrite this and couldn't come up with anything I liked. If we think of anything better we can change it.